core/src/main/java/tc/oc/pgm/map/MapInfoImpl.java

```diff
@@ -243,6 +254,18 @@ public Component getStyledName(MapNameStyle style) {
     return rules;
   }
 
+  private static List<Gamemode> parseGamemodes(Element root) throws InvalidXMLException {
+    List<Gamemode> gamemodes = new ArrayList<>();
+    for (Element gamemodeEl : root.getChildren("gamemode")) {
+      for (Gamemode gamemode : Gamemode.values()) {
+        if (gamemode.getId().equals(gamemodeEl.getText())) {
```

Comment:
Consider moving this to a static method in gamemode, so that you can call it by doing:
```
Gamemode gm = Gamemode.byId(gamemodeEl.getText());
if (gamemode == null) throw new InvalidXMLException("Unknown gamemode", gamemodeEl);
else gamemodes.add(gamemode);
```

Currently if the gamemode doesn't exist you're just ignoring it, it's an error that needs to be reported or else you may set your map to be "wools" (instead of ctw), assume that's valid, never check-back, and pgm will never say a thing about it.

Also test what happens when you just define a `<gamemode/>` tag without any text, does it blow up, or does it simply complain?